code401Challenges/src/main/java/code401Challenges/hashtable/LeftJoin.java

```diff
@@ -1,36 +1,26 @@
 package code401Challenges.hashtable;
 
 import java.util.ArrayList;
+import java.util.LinkedList;
 
 public class LeftJoin {
 
     public ArrayList<String[]> leftJoinTwoHashmaps(Hashtable hashtable1, Hashtable hashtable2) {
-        ArrayList<String[]> leftJoinArr = new ArrayList<>();
-        ArrayList<String> keys = new ArrayList<>();
-
-        // iterate through hashtable1 and put its key in the arrayList
-        // <------------------------------------------------------ come back to this
-
-        // iterate through keys in keys ArrayList, ie ["fond", "wrath", "diligent"]
-        for (int i = 0; i < keys.size(); i++) {
-            // create empty array of size 3 <----------- does this re-empty the array each iteration?
-            String[] tempArr = new String[3];
-
-            if(hashtable1.contains(keys.get(i))) { // <------- check that these gets are grabbing correct thing
-                // add the key to the tempArr to index 0
-                tempArr[i] = keys.get(i);
-                // add the value to the tempArr to index 1
-                tempArr[i+1] = hashtable1.get(keys.get(i));
-                // add the value in hashmap2 to index 3 if value exists in hashtable2
-
-                if(hashtable2.contains(keys.get(i))){
-                    tempArr[i+2] = hashtable2.get(keys.get(i));
-                } else {
-                    tempArr[i+2] = null;
+        ArrayList<String[]> answer = new ArrayList<>();
```

Comment:
You might consider making this a `static` method so that you don't have to instantiate LeftJoin objects (which feels a bit awkward) in order to use this method. If you make it static then when it comes time to use the method (like in your tests) you just write: `LeftJoin.leftJoinTwoHashmaps(ht1, ht2);`